### PR TITLE
test: improve e2e testing using rust 

### PR DESCRIPTION
**Pull request by fabriziosestito**
## Description

Supersedes: https://github.com/kubewarden/kwctl/pull/594
by rewriting all the existing tests and adding tests
for `pull`, `scaffold mainfest`

Fixes: #602 

## Test

yes

## Additional Information
rstest  was used to dry-up similar test cases where possible.
some trivial test was removed.

### CODE REVIEW

**Review by jvanz — APPROVED** [review]

File: tests/secure_supply_chain_e2e.rs [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+#[test]
+fn test_verify_tuf_integration() {
+    let tempdir = tempdir().unwrap();
+    let mut cmd = setup_command(tempdir.path());
+
+    cmd.arg("verify")
+        .arg("--verification-config-path")
+        .arg(test_data("sigstore/verification-config-keyless.yml"))
+        .arg("registry://ghcr.io/kubewarden/policies/capabilities-psp:v0.1.9");
```
Comment (viccuad) [inline tests/secure_supply_chain_e2e.rs:24]: We should be using policies from `ghcr.io/kubewarden/tests`.

File: tests/secure_supply_chain_e2e.rs [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+    cmd.arg("verify")
+        .arg("--fulcio-cert-path")
+        .arg(".sigstore/root/targets/fulcio.crt.pem")
+        .arg("--fulcio-cert-path")
+        .arg(".sigstore/root/targets/fulcio_v1.crt.pem")
+        .arg("--rekor-public-key-path")
+        .arg(".sigstore/root/targets/rekor.pub")
+        .arg("--verification-config-path")
+        .arg(test_data("sigstore/verification-config.yml"))
+        .arg("registry://ghcr.io/kubewarden/policies/capabilities-psp:v0.1.9");
```
Comment (viccuad) [inline tests/secure_supply_chain_e2e.rs:111]: We should be using policies from `ghcr.io/kubewarden/tests`.

File: tests/secure_supply_chain_e2e.rs [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
```
Comment (viccuad) [inline tests/secure_supply_chain_e2e.rs:133]: We should be using policies from `ghcr.io/kubewarden/tests`.

File: tests/secure_supply_chain_e2e.rs [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+    cmd.arg("verify")
+        .arg("--fulcio-cert-path")
+        .arg(".sigstore/root/targets/fulcio.crt.pem")
+        .arg("--fulcio-cert-path")
+        .arg(".sigstore/root/targets/fulcio_v1.crt.pem")
+        .arg("--rekor-public-key-path")
+        .arg(".sigstore/root/targets/rekor.pub")
+        .arg("--verification-config-path")
+        .arg(&verification_config_path)
+        .arg("registry://ghcr.io/kubewarden/policies/capabilities-psp:v0.1.9");
```
Comment (viccuad) [inline tests/secure_supply_chain_e2e.rs:168]: We should be using policies from `ghcr.io/kubewarden/tests`.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -133,15 +129,15 @@ jobs:
         run: |
           ./scripts/kubewarden-load-policies.sh \
               --policies policies.tar.gz \
-              --policies-list e2e-tests/airgap/policies.txt \
+              --policies-list tests/data/airgap/policies.txt \
```
Comment (viccuad) [inline .github/workflows/ci.yml:132]: Given that now we use testcontainers, we could move this airgap test out of GHA and into local rust tests.
But this is not a merge blocker for me.

**Review by viccuad — CHANGES_REQUESTED** [review]
LGTM, this is great!
Just one requested change, to move to policies in ghcr.io/kubewarden/tests, on tests/secure_supply_chain_e2e.rs.

File: .github/workflows/ci.yml (hunk 5 above)
Comment (fabriziosestito) [reply .github/workflows/ci.yml:132]: @viccuad I will add an issue about this

**Review by viccuad — APPROVED** [review]
Many thanks for the fix, LGTM!

**Pull request merged**